app/src/main/java/com/example/parkingsystem/mvp/model/ReleaseModel.java

```diff
@@ -37,17 +43,30 @@ public int numberOfMatchesOfTheReservation(Reservation reservationToBeReleased)
         return numberOfMatches;
     }
 
-    public boolean releaseParking(Reservation newReservation) {
-        if (getParking().getReservationsList().size() == 0 ||
-                numberOfMatchesOfTheReservation(newReservation) > 1) {
-            return false;
+    public void releaseParking(Reservation newReservation) throws ReleaseEmptyListException, ReleaseMoreThanOneMatchException, ReleaseNoMatchesException {
+        if (getParking().getReservationsList().size() == 0) {
+            throw new ReleaseEmptyListException();
         }
-        for (Reservation reservation : getParking().getReservationsList()) {
-            if (reservation.equals(newReservation)) {
-                getParking().getReservationsList().remove(reservation);
-                return true;
+        if (numberOfMatchesOfTheReservation(newReservation) > 1) {
+            throw new ReleaseMoreThanOneMatchException();
+        }
+        if (numberOfMatchesOfTheReservation(newReservation) == 1) {
+            for (Reservation reservation : getParking().getReservationsList()) {
+                if (reservation.equals(newReservation)) {
+                    getParking().getReservationsList().remove(reservation);
+                    return;
+                }
             }
```

Comment:
This FOR loop is not needed, you can remove it and just do `getParking().getReservationsList().remove(reservation);`